fa-etl.py

```diff
@@ -456,30 +480,36 @@ def join(input_dir, ranked_valhist_filename, prop_filename, ranked_deed_filename
 
     logging.info(f"Joining ranked val hist to prop, ranked_deed, and taxhist...")
     # https://docs.pola.rs/py-polars/html/reference/lazyframe/api/polars.LazyFrame.join.html
-    (ranked_valhist.join(
+    step1 = (ranked_valhist.join(
         # first join in the data from the annual file (prop characteristics)
         other= prop,
         how = "left",
         on='PropertyID',
         #validate='m:1', #checks if only 1 propertyid in annual file
-        force_parallel=True 
+        # force_parallel=True 
         # second join in the data from the most recent sale of each year
-        ).join(
-            other=ranked_deed,
-            how='left',
-            left_on=['PropertyID', 'Year'], 
-            right_on=['PropertyID','RecordingYear']
-        ).join(
-            other=taxhist,
-            how='left',
-            left_on=['PropertyID', 'Year'], 
-            right_on=['PropertyID','TaxYear']
+        )).collect(streaming=True)
```

Comment:
@nmarchio hi nico! I'm still hitting a memory error (exit code 137) when attempting this join. 

my read of your code is that joins are better with collects (vs. sinks), so i'm trying this implementation but still hitting memory issues. do you have any other ideas as to why this would still be failing? happy to jump on a call in the morning if helpful!